### PR TITLE
Header and announcement list UI refactors

### PR DESCRIPTION
**Pull request by mborsare**
This PR is concerned mostly with Constable's header. It refactors the UI for more order. It gives the search feature more prominence to help readers find announcements easier. It also tightens up the vertical space previously used by the widget to access one's subscribed interests. Now that anchors the right side of the navigation container. Lastly, it reworks the small screen header design so everything has ample living space. Previously, the elements felt squished together on small screens.

In addition this PR makes a few changes to the announcement list. First, it scales back the use of red for the announcement list headlines. Red is a powerful color and we want to reserve that power for things like buttons, *inline links* or other strong actions. Lastly, it makes the avatars a little smaller and increases the size of the headlines for improved hierarchy.

fuego

### CODE REVIEW

File: assets/css/_header.scss [hunk 1]
```diff
@@ -1,50 +1,103 @@
...
+  position: relative;
+
+  @include media($medium-screen-up) {
+    flex-basis: 18.75rem;
+    order: 2;
+  }
+}
+
+.header-search-form::before {
+  @include position(absolute, 28% null null 20%);
```
Comment (enatario) [inline assets/css/_header.scss]: I feel like this lil guy should be locked to the left side of the input (with a slight offset so it looks like it's inside like now). It was a slightly jarring for it to move around on resize.

File: lib/constable_web/templates/layout/app.html.eex [hunk 2]
```diff
@@ -19,33 +19,27 @@
...
-              />
-            <% end %>
-          </li>
-        </ul>
+        <%= form_tag search_path(@conn, :show), class: "header-search-form", method: :get do %>
+          <input
+            class="header-search-input"
+            name="query"
+            placeholder="Search Constable..."
+            type="text"
```
Comment (enatario) [inline lib/constable_web/templates/layout/app.html.eex]: Not sure if this'll mess with the functionality at all, but the type on this should be `search`

File: lib/constable_web/templates/layout/app.html.eex [hunk 3]
```diff
@@ -19,33 +19,27 @@
...
-            <%= link to: settings_path(@conn, :show), class: "settings-gravatar remote-append" do %>
-              <img
-                src="<%= gravatar(@conn.assigns.current_user) %>"
-                class="current-user-gravatar"
-              />
-            <% end %>
-          </li>
-        </ul>
+        <%= form_tag search_path(@conn, :show), class: "header-search-form", method: :get do %>
+          <input
```
Comment (enatario) [inline lib/constable_web/templates/layout/app.html.eex]: I know this is legacy code, but might be good to add a corresponding label with this guy with a "for" attribute that links up with an id on the search for ~ accessibility ~. You can just hide it visually so it doesn't show up.

Comment (paulcsmith) [reply lib/constable_web/templates/layout/app.html.eex]: > legacy code

sad

Comment (enatario) [reply lib/constable_web/templates/layout/app.html.eex]: 😂

**Comment by mborsare** [comment]
@paulcsmith I'll experiment with bringing the avatars up in size some. I'll also build in an affordance for that horizontal scroll. Good idea!

File: lib/constable_web/templates/layout/app.html.eex (hunk 2 above)
Comment (mborsare) [reply lib/constable_web/templates/layout/app.html.eex]: haha that totally busted things 🙉

Comment (mborsare) [reply lib/constable_web/templates/layout/app.html.eex]: oh n/m it's just a selector issue sheewwww

Comment (enatario) [reply lib/constable_web/templates/layout/app.html.eex]: lolol

re: `sheewwww` i just made that noise in my head as i read it

**Comment by mborsare** [comment]
fuego-updates

**Comment by paulcsmith** [comment]
That looks nice! Would it be possible to center the navigation when it wraps?

**Comment by mborsare** [comment]
@paulcsmith yup, I'll do that and merge it up

**Comment by paulcsmith** [comment]
Sounds good :D I’m stoked to see this live

**Pull request merged**